### PR TITLE
Added description of listener hirarchy and updated status table [9040]

### PR DESCRIPTION
**Pull request by imontesino**

### CODE REVIEW

File: docs/fastdds/dds_layer/core/entity/entity.rst [hunk 1]
```diff
@@ -89,6 +89,19 @@ Then they can link these listeners to each entity, either during their creation
 |DataReader::set_listener-api|).
 The listener interfaces that each entity type and their callbacks are explained in the documentation
 for each entity type.
+In the case in which the listeners of different entities have the same callback defined, the order in which each
+callback is executed is dictated by the inheritance from parent to child, i.e. from the lowest level entity to the
+highest.
```
Comment (IkerLuengo) [inline docs/fastdds/dds_layer/core/entity/entity.rst]: This needs rewording. Is not only when the listener has the callback defined, but that any event is handled by the lowest-level entity that:
- has a not-null listener attached
- and the specific callback is enabled on the StatusMask
Also, `the order in which each callback is executed` suggests that all listeners are triggered one after the other.

File: docs/fastdds/dds_layer/core/entity/entity.rst [hunk 2]
```diff
@@ -107,8 +108,8 @@ highest.
 
 Status
 ^^^^^^
-Each entity is associated with a set of status objects whose values represent the *communication status*
 of that entity.
+Each entity is associated with a set of status objects whose values represent the *communication status*
```
Comment (IkerLuengo) [inline docs/fastdds/dds_layer/core/entity/entity.rst:112]: Look like this line was moved unintentionally

File: docs/fastdds/dds_layer/core/entity/entity.rst [hunk 3]
```diff
@@ -89,9 +89,10 @@ Then they can link these listeners to each entity, either during their creation
 |DataReader::set_listener-api|).
 The listener interfaces that each entity type and their callbacks are explained in the documentation
 for each entity type.
-In the case in which the listeners of different entities have the same callback defined, the order in which each
-callback is executed is dictated by the inheritance from parent to child, i.e. from the lowest level entity to the
-highest.
+When an event occurs it is handled by the lowest level entity with a listener that is non-null
+and has the corresponding callback enabled in its |StatusMask-api|.
+Higher level listeners inherit from the lower level ones as shown in the following
+diagram:
```
Comment (IkerLuengo) [inline docs/fastdds/dds_layer/core/entity/entity.rst:95]: :+1:

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/603/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/604/)

**Pull request merged**